pkg/minikube/cluster/cluster.go

```diff
@@ -680,3 +681,51 @@ func EnsureMinikubeRunningOrExit(api libmachine.API, exitStatus int) {
 		os.Exit(exitStatus)
 	}
 }
+
+type ServiceURL struct {
+	Namespace string
+	Name      string
+	URL       string
+}
+
+type ServiceURLs []ServiceURL
+
+func GetServiceURLs(api libmachine.API, namespace string, t *template.Template) (ServiceURLs, error) {
+	host, err := CheckIfApiExistsAndLoad(api)
+	if err != nil {
+		return nil, err
+	}
+
+	ip, err := host.Driver.GetIP()
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := getKubernetesClient()
+	if err != nil {
+		return nil, err
+	}
+
+	getter := getKubernetesServicesWithNamespace(client, namespace)
+
+	svcs, err := getter.List(kubeapi.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var serviceURLs []ServiceURL
+
+	for _, svc := range svcs.Items {
+		url, err := getServiceURLWithClient(client, ip, svc.Namespace, svc.Name, t)
```

Comment:
An error would be pretty unexpected in this case so kinda think it's OK to return an error as it's likely to point to a more fundamental issue.